eclair-core/src/main/scala/fr/acinq/eclair/transactions/CommitmentSpec.scala

```diff
@@ -42,6 +42,34 @@ object CommitmentOutput {
 
 }
 
+sealed trait DirectedPtlc {
```

Comment:
In my opinion, there should be a super class that is shared between `DirectedPtlc` and `DirectedHtlc`. Let's called it `CommitmentScript`. 

```scala
sealed trait CommitmentScript //maybe we can extend org.bitcoins.core.protocol.script.Script ???

sealed trait DirectedHtlc extends CommitmentScript

sealed trait DirectedPtlc extends CommitmentScript 
```

_Theoretically_ (dangerous word :P) then we can start replacing all of the `DirectedHTLC` throughout the codebase with `CommitmentScript` and see what he damage is. 

Let's also take a look at things like `CommitmentSpec.addHtlc()`, this should probably be renamed to `CommitmentSpec.addCommitmentScript()` or something like that

```scala
def addCommitmentScript(spec: CommitmentSpec, directedScript: CommitmentScript): CommitmentSpec
```

We _hopefully_ can use Scala's pattern match exhaustiveness properties to get nice compiler errors/warnings on where things need to be updated. This is important to get done, because as we add _more_ `CommitmentScript` types, the compiler will tell us where we need to update our code :-)

Does this make sense?

Reply:
`update_add_ptlc` is not a script, it's a LN protocol channel update message. I'd like to create a parallel class hierarchies for all PTLC related stuff to avoid nasty merge conflicts and to localize PTLC changes.

Reply:
We can always refactor to merge the parallel hierarchies/code when/if this implementation will be actually adopted.